Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -486,10 +509,15 @@ DOM::ExceptionOr<void> XMLHttpRequest::send(Optional<XMLHttpRequestBodyInit> bod
 
     auto request = LoadRequest::create_for_url_on_page(request_url, m_window->page());
     request.set_method(m_method);
-    if (!body_with_type.body.is_empty()) {
-        request.set_body(body_with_type.body);
-        if (!body_with_type.type.is_empty())
-            request.set_header("Content-Type", body_with_type.type);
+    if (body_with_type.has_value()) {
+        body_with_type->body.source().visit(
+            [&](Empty) {},
+            [&](ByteBuffer const& buffer) {
+                request.set_body(buffer);
+            },
+            [&](NonnullRefPtr<FileAPI::Blob> const&) {});
+        if (body_with_type->type.has_value())
+            request.set_header("Content-Type", String { ReadonlyBytes { body_with_type->type.release_value() } });
```

Comment:
- No need to `release_value` if we're only reading from it anyway
- We can use `.span()` to get `ReadonlyBytes`

```suggestion
            request.set_header("Content-Type", String { body_with_type->type->span() });
```